lib/charty/table_adapters/hash_adapter.rb

```diff
@@ -104,7 +104,12 @@ def initialize(data, columns: nil, index: nil)
         arrays.each do |array|
           index = case array
                   when Charty::Vector
-                    array.index
+                    case array.index
+                    when DaruIndex, PandasIndex
+                      Index.new(array.index.to_a)
+                    else
+                      array.index
+                    end
```

Comment:
We can't use `DaruIndex` and `PandasIndex` for `Array` data.d

Reply:
I want to know several concrete cases that are problematic to use `DaruIndex` or `PandasIndex` as is.
Could you show example cases?

I want to prevent converting `Daru::Vector` and `Pandas::Vector` to an `Array` as possible.

Reply:
OK.

For example, https://github.com/red-data-tools/charty/blob/79e9406940fe89ddb94bd1f9b01ad9bda17210b0/lib/charty/table_adapters/hash_adapter.rb#L162 expects `#size` but `Arrow::Array#size` doesn't exist. (`Arrow::Array#length` exists.)

How about using `Charty::Vector` for elements of `@data` of `Charty::TableAdapters::HashAdapter` instead of using `Charty::Vector#data` value?

```diff
diff --git a/lib/charty/table_adapters/hash_adapter.rb b/lib/charty/table_adapters/hash_adapter.rb
index 47f12bb..79414fc 100644
--- a/lib/charty/table_adapters/hash_adapter.rb
+++ b/lib/charty/table_adapters/hash_adapter.rb
@@ -121,16 +121,7 @@ module Charty
         index = union_indexes(*indexes)
 
         arrays = arrays.map do |array|
-          case array
-          when Charty::Vector
-            array.data
-          when Hash
-            raise NotImplementedError
-          when self.class.method(:array?)
-            array
-          else
-            Array.try_convert(array)
-          end
+          Vector.try_convert(array)
         end
 
         columns = generate_column_names(arrays.length, columns)
```

Reply:
I see. Could you please make the change to do so?

Reply:
Done.

Reply:
We need many changes for this...